cors.go

```diff
@@ -87,7 +87,9 @@ type Cors struct {
 	// Set to true when allowed headers contains a "*"
 	allowedHeadersAll bool
 	allowCredentials  bool
-	optionPassthrough bool
+	// OptionsPassthrough indicates if preflight lets other potential next
+	// handlers to process the OPTIONS method.
+	OptionPassthrough bool
```

Comment:
I would prefer not to expose this. Couldn't we create a wrapping type in the gin wrapper package to keep the knowledge of this option in the gin wrapper side without having to access it from the actual cors.Cors type?

```go
type corsWrapper struct {
    cors.Cors
    optionPassthrough bool
}

func New(options Options) gin.HandlerFunc {
    return corsWrapper{cors.New(options), options.OptionPassthrough}
}

// implement gin.Handler interface on corsWrapper
```

Reply:
@rs yes, but that won't work if we'll try to wrap and existing `cors.Cors` handler with `Wrap()` (it transforms any `*cors.Cors` to `gin.HandlerFunc`).

Another option I see here is to make a public method:
```go
func (c *Cors) IsPassthroughtEnabled() bool {
    return c.optionPassthrough
}
```

That will disallow `c.optionPassthrough` mutation.

Reply:
@rs 

>>>
In what context would you need to wrap an existing cors?
>>>

That may be useful for someone reusing/extending existing code.

However that's not an wide spread case. And even then someone can reuse options, though.

Ok. Then we just can hide `Wrap()` method.